src/com/seafile/seadroid2/account/AccountManager.java

```diff
@@ -67,6 +67,10 @@ public Account getAccountBySignature(String signature) {
      * @param account
      */
     public void saveAccountToDB(Account account) {
+        Account act = dbHelper.getAccount(account.getServer(), account.getEmail());
+        if (act != null)
+            return;
```

Comment:
I made a wrong comment previously. AccountDBHeloper.saveAccount already checks if there is an existing account, so we don't need to double chekc here.